code/game/machinery/doors/shutters.dm

```diff
@@ -15,3 +17,36 @@
 /obj/machinery/door/poddoor/shutters/indestructible
 	name = "hardened shutters"
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+
+/obj/machinery/door/poddoor/shutters/radiation
+	name = "radiation shutters"
+	desc = "Lead-lined shutters with a raddiation hazard symbol on it. Once closed, this blocks some radiation."
+	icon = 'icons/obj/doors/shutters_radiation.dmi'
+	icon_state = "closed"
+	rad_insulation = 0.2
+
+/obj/machinery/door/poddoor/shutters/radiation/preopen
+	icon_state = "open"
+	density = FALSE
+	opacity = 0
+	rad_insulation = 1
+
+/obj/machinery/door/poddoor/shutters/radiation/do_animate(animation)
+	. = ..()
+	switch(animation)
+		if("opening")
+			rad_insulation = 1
+		if("closing")
+			rad_insulation = -0.5	//Rad insulation 0 to 1 doesn't really do anything to radiation, above 1 starts multiplying it, you need to go into negative numbers to start blocking radiation.
```

Comment:
This code needs to go into the opening and closing procs, instead of the animation one.

Reply:
It now does this on open() and close().
Tested and working as intended.
![image](https://user-images.githubusercontent.com/53862927/93707732-0d160800-fb29-11ea-99a6-9f5c853d7f46.png)
![image](https://user-images.githubusercontent.com/53862927/93707736-18693380-fb29-11ea-96ac-8c31c595a458.png)